### ISSUE TITLE
Enum false positives

### ISSUE THREAD

**Issue by NorfairKing** [ocharles/weeder#148]
If you have

```
data Alphabet = A | B | C
  deriving (Enum, Bounded)
entireAlphabet :: [Alphabet]
entireAlphabet = [minBound .. maxBound]
```

and you set `entireAlphabet` as a root(reached), then the `Enum` instance will be marked as a weed, eventhough ghci says:

```
ghci> :t [minBound .. maxBound]
[minBound .. maxBound] :: (Bounded a, Enum a) => [a]
```

**Comment by ocharles** [comment]
Yes, unfortunately this idiom doesn't produce enough evidence usage. @ryndubei did we ever report this one to GHCs tracker?

**Comment by ryndubei** [comment]
Yes, this currently the case with most overloaded syntax. The default configuration currently only marks `IsString` and `IsList` as roots because the instances of these classes are the most likely to only show up in the form of overloaded syntax, but it can be a problem for even `Eq` if it is only used in pattern matches, for example.

This should be fixed in GHC 9.10.

**Comment by ocharles** [comment]
Awesome, thanks! On that I'm inclined to close this - what do you think @NorfairKing? We do mention this in the README - https://github.com/ocharles/weeder?tab=readme-ov-file#overloaded-syntax. Do you think that needs more content?

**Comment by NorfairKing** [comment]
Cool, closing this.
@ocharles You could mention that Enum is also relevant for that caveat, and maybe that it will be fixed in GHC 9.10 as well.
